### ISSUE TITLE
Why normalizing the scaled feature map?

### ISSUE THREAD

**Issue by jordipons** [marl/autopool#2]
I don't fully grasp the need of this normalization that you employ in your code:
`softmax = tf.keras.backend.exp(scaled - max_val)`

I'm tempted to remove this line from my implementation.
Any ideas on that? Does it help training?

Jordi.

**Comment by bmcfee** [comment]
You definitely should not remove that line.  This is a standard improvement on softmax calculation that enhances numerical stability; see https://stackoverflow.com/questions/42599498/numercially-stable-softmax

**Comment by jordipons** [comment]
Thanks, Brian! I didn't know this improvement.
